### PR TITLE
Basic Tenant deployment script

### PR DESCRIPTION
**Pull request by dvaldivia**
Basic script to test a tenant can be created

Signed-off-by: Daniel Valdivia <18384552+dvaldivia@users.noreply.github.com>

### CODE REVIEW

File: testing/basic-tenant.sh [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+echo "Installing lite tenant"
+
+try kubectl apply -k ../examples/kustomization/tenant-lite
+
+
+echo "Waiting for the tenant statefulset, this indicates the tenant is being fulfilled"
+waitdone=0
+totalwait=0
+while :; do
```
Comment (harshavardhana) [inline testing/basic-tenant.sh:36]:
```suggestion
while true; do
```

Let's use readable values.

File: testing/basic-tenant.sh [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  --selector="v1.min.io/tenant=tenant-lite" \
+  --timeout=300s
+
+echo "Build passes basic tenant creation"
+
+# Check MinIO is accessible
+
+waitdone=0
+totalwait=0
+while :; do
```
Comment (harshavardhana) [inline testing/basic-tenant.sh:62]:
```suggestion
while true; do
```

File: testing/basic-tenant.sh [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  waitdone=$(kubectl -n tenant-lite get pods -o json | jq '.items | length')
+  if [ "$waitdone" -ne 0 ]; then
+    echo "Found $waitdone pods"
+    break
+  fi
+  sleep 5
+  totalwait=$((totalwait + 5))
+  if [ "$totalwait" -gt 300 ]; then
```
Comment (harshavardhana) [inline testing/basic-tenant.sh:70]:
```suggestion
  if [ "$totalwait" -gt 305 ]; then
```

File: testing/basic-tenant.sh [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    echo "Tenant never created statefulset after 5 minutes"
```
Comment (harshavardhana) [inline testing/basic-tenant.sh:71]:
```suggestion
    echo "Unable to create tenant after 5 minutes, exiting."
```

File: testing/basic-tenant.sh [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  --timeout=300s
+
+echo "Build passes basic tenant creation"
+
+# Check MinIO is accessible
+
+waitdone=0
+totalwait=0
+while :; do
+  waitdone=$(kubectl -n tenant-lite get pods -o json | jq '.items | length')
```
Comment (harshavardhana) [inline testing/basic-tenant.sh:63]:
```suggestion
  waitdone=$(kubectl -n tenant-lite get pods -l v1.min.io/tenant=tenant-lite --no-headers | wc -l)
```

isn't this simpler?

File: testing/basic-tenant.sh [hunk 6]
```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+# This script requires: kubectl, kind, jq
+
+#set -e
+
+yell() { echo "$0: $*" >&2; }
+die() {
```
Comment (harshavardhana) [inline testing/basic-tenant.sh:8]:
```suggestion

die() {
```

File: testing/basic-tenant.sh [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+  yell "$*"
+  (kind delete cluster || true ) && exit 111
+}
+try() { "$@" || die "cannot $*"; }
```
Comment (harshavardhana) [inline testing/basic-tenant.sh:12]:
```suggestion

try() { "$@" || die "cannot $*"; }
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: testing/basic-tenant.sh [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+  if [ "$totalwait" -gt 300 ]; then
+    echo "Tenant never created statefulset after 5 minutes"
+    try false
+  fi
+done
+
+echo "Waiting for tenant pods to come online (5m timeout)"
+try kubectl wait --namespace tenant-lite \
+  --for=condition=ready pod \
+  --selector="v1.min.io/tenant=tenant-lite" \
```
Comment (cniackz) [inline testing/basic-tenant.sh:55]: This selector is not working, by looking at the pod's labels, we should be using below selector:
`--selector="v1.min.io/tenant=storage-lite"`
Notice this time is using "storage" and not "tenant" 
I got the labels with command below:
`kubectl get pods --show-labels -n tenant-lite`
And I assume, we want to wait for these pods:

- storage-lite-ss-0-0
- storage-lite-ss-0-1
- storage-lite-ss-0-2
- storage-lite-ss-0-3

File: testing/basic-tenant.sh [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+  --timeout=300s
+
+echo "Build passes basic tenant creation"
+
+# Check MinIO is accessible
+
+waitdone=0
+totalwait=0
+while true; do
+  waitdone=$(kubectl -n tenant-lite get pods -l v1.min.io/tenant=tenant-lite --no-headers | wc -l)
```
Comment (cniackz) [inline testing/basic-tenant.sh:65]: The same applies for `-l v1.min.io/tenant=tenant-lite` where the selector is wrong.
It should be `-l v1.min.io/tenant=storage-lite`

**Review by cniackz — CHANGES_REQUESTED** [review]
The same applies for `-l v1.min.io/tenant=tenant-lite` where the selector is wrong.
It should be `-l v1.min.io/tenant=storage-lite` as per the node label.

File: testing/basic-tenant.sh (hunk 8 above)
Comment (cniackz) [reply testing/basic-tenant.sh]: Added new commit on top of this pull request to solve the issue

File: testing/basic-tenant.sh (hunk 9 above)
Comment (cniackz) [reply testing/basic-tenant.sh]: Added new commit on top of the pull request to solve the problem

**Review by cniackz — DISMISSED** [review]
The script is now working when is executed

File: .github/workflows/bash.yml [hunk 10]
```diff
@@ -0,0 +1,33 @@
+# This is a basic workflow to execute our Tenant deployment script
+
+name: Bash
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches: [ main ]
```
Comment (dvaldivia) [inline .github/workflows/bash.yml:9]:
```suggestion
    branches: [ master ]
```

File: .github/workflows/bash.yml [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+  pull_request:
+    branches: [ main ]
```
Comment (dvaldivia) [inline .github/workflows/bash.yml:11]:
```suggestion
    branches: [ master ]
```

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**